src/include/storage/recovery/replication_log_provider.h

```diff
@@ -1,20 +1,143 @@
 #pragma once
```

Comment:
I would like to take a second pass on this file when I'm fresh. This is a "here be dragons" file to me that I haven't looked at much / there may be opportunities for refactoring here.

Reply:
I will probably take a pass on this tonight after submitting some other hw @jkosh44 maybe don't look at this until tomorrow Friday, thanks!